store.go

```diff
@@ -578,8 +578,8 @@ type AdditionalLayer interface {
 	// layer store.
 	PutAs(id, parent string, names []string) (*Layer, error)
 
-	// UncompressedDigest returns the uncompressed digest of this layer
-	UncompressedDigest() digest.Digest
+	// TOCDigest returns the digest of TOC of this layer. Returns "" if unknown.
+	TOCDigest() digest.Digest
```

Comment:
Compare https://github.com/containers/image/pull/2416#discussion_r1606886998 / https://github.com/containerd/stargz-snapshotter/pull/1673#discussion_r1607020173 ; do we need this provided by the ALS backend at all?

Reply:
… we actually, sort of, do: if this returns `""`, we know the FUSE server is older than this PR expects, and interprets the digest as non-TOC compressed digest.

AFAICS the proposed `stargz-snapshotter` implementation never returns a _different_ digest than the one we looked up, but at least the “supported/unsupported” bit seems valuable.